tests/wcag/wcagreporter.js

```diff
@@ -0,0 +1,147 @@
+const { BrowserPageWidths } = require("../percy/constants");
+
+class WcagReporter {
+  /**
+   * @param {PageNavigator} pageNavigator
+   * @param {import('@axe-core/puppeteer')} analyzer
+   * @param {import('puppeteer').Page} page
+   */
+  constructor (pageNavigator, analyzer, page) {
+    this._pageNavigator = pageNavigator;
+    this._analyzer = analyzer;
+    this._page = page;
+    this.results = [];
+  }
+
+  async analyze() {
+    await this._analyzeUniversalSearch();
+    await this._analyzeVerticalSearch();
+    await this._analyzeVerticalGridSearch();
+    await this._analyzeVerticalMapSearch();
+    await this._analyzeVerticalFullPageMapSearch();
+    await this._analyzeDirectAnswers();
+    return this.results;
+  }
+
+
+  async _analyzeUniversalSearch() {
```

Comment:
This seems to mimic the exact flow of the `Photographer` class. I think we should do some re-factoring here. For both classes, we go to the same pages, run the same queries, etc. If we add a new page or want to run a new query, we will have to update both files.

Reply:
Maybe we move towards a CAC approach? Something like:

```
interface TestCommand {
  type: 'click',
  params?: any[]
}

interface TestLocation {
  pageType: 'vertical' | 'universal',
  page?: string
  name: string
  viewPort?: {
    height?: number,
    width?: number
  }
  commands?: TestCommand[]
}

const testingLocations : TestLocation[] = [
  {
    name: 'universal-search',
    pageType: 'universal'
  },
  {
    name: 'vertical-grid-search',
    pageType: 'vertical',
    page: 'people'
  },
  {
    name: 'universal-search-faq-accordion',
    pageType: 'universal',
    commands: [
      { type: 'click', params: ['.HitchhikerFaqAccordion-toggle']}
    ]
  }
]
```

Reply:
I see, that makes sense. Although we wouldn't have nice interface/typing since theme is only in js right now. Yeah I think  having a PageOperator that runs through these test configs would be good.

Reply:
How would this `TestLocation[]` fit into what we have currently? Would the `Photographer` and `WCAGReporter` classes take them in? Then, for each entry in the array, the respective class would know what action to take? Another approach is to simply have a class that does all the navigating, clicking, etc. and takes the snapshot and does the WCAG analysis. For example: 

```
    await this._pageNavigator.gotoUniversalPage({ query: 'what if i forget my password?'});
    await this._pageNavigator.click('.HitchhikerFaqAccordion-toggle')
    await this._camera.snapshot('universal-search--faq-accordion');
    this.results.push(await this._analyzer.analyze());
```
This approach is fairly straightforward and we don't need to create a CaC framework.

Reply:
I think the CaC approach is pretty neat, and it might also help us out with the multilangphotographer's tests as well. 
I imagine it's something like this:
```
while(pageOperator.hasNextTestLocation()) {
   const testName = pageOperator.nextTestLocation();
   this._camera.snapshot(testName); OR this.results.push(await this._analyzer.analyze());
}
```
I already have the pageOperator code that would parse the file and run the commands, but yes there would need to be more changes in other files and writing up the testLocations itself.
Perhaps, we can keep this PR as is for the format, and make a new item to address the CaC or page navigation refactoring in general?